src/core/xref.js

```diff
@@ -71,8 +71,8 @@ export async function run(conf) {
   const queryKeys = [];
   for (const elem of elems) {
     const entry = getRequestEntry(elem);
-    const id = await objectHash(entry);
-    queryKeys.push({ ...entry, id });
+    entry.id = await objectHash(entry);
```

Comment:
This will mean now the objectHash also hashes `id: ""`, but it's ok as we just need a checksum ID for caching.